context/index.d.ts

```diff
@@ -0,0 +1,17 @@
+import type { ReadableAtom } from '../atom'
+
+// This isn't a part of the public API for now
+type Context = unknown
+
+export type ContextGetter = <T extends ReadableAtom>(store: T) => T
+export type WithCtx = {
+  ctx: ContextGetter
+}
+
+export declare const globalContext: Context
+
+export function createContext(storeValues?: object): Context
```

Comment:
We will need better types for that to be able to use it in mocks.

I think we need to plan context API for mocks and tests on the current state of PR.